lib/datasource.js

```diff
@@ -513,7 +513,12 @@ DataSource.prototype.setupDataAccess = function(modelClass, settings) {
       idProp.type = idType;
       modelClass.definition.rawProperties[idName].type = idType;
       modelClass.definition.properties[idName].type = idType;
-      if (settings.forceId) {
+      var forceId = settings.forceId;
+      if (['cloudant', 'mongodb'].indexOf(this.connector.name) === -1 &&
```

Comment:
Please don't hard-code connector names in juggler. It inverts dependency direction and is difficult to scale.

You should check if the `id` column is auto-incremented instead, that's connector agnostic. I think we should set `forceId` whenever the `id` field is generated by the database in whatever way, not only when it's an incrementing sequence.

In my understanding, `idProp.generated` is exactly the flag we are looking for. Note that this flag is set for non-SQL connectors (like MongoDB) too. I think it makes sense, because for example our memory connector uses auto-incremented numeric ids even though it's considered as NoSQL connector.

Alternatively, if we are not comfortable setting `forceId` for all generated id properties, then we can limit it only to generated id properties of type Number. That will cover SQL & memory, but exclude MongoDB.